src/app/parser/AbstractLabradorParser.ts

```diff
@@ -0,0 +1,140 @@
+//
+// LESERKRITIKK v2 (aka Reader Critics)
+// Copyright (C) 2017 DB Medialab/Aller Media AS, Oslo, Norway
+// https://github.com/dbmedialab/reader-critics/
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later
+// version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+import * as Cheerio from 'cheerio';
+
+import ArticleAuthor from 'base/ArticleAuthor';
+
+import AbstractIteratingParser from 'app/parser/AbstractIteratingParser';
+import IteratingParserItem from 'app/parser/IteratingParserItem';
+
+import { getOpenGraphAuthors } from 'app/parser/util/AuthorParser';
+import { getOpenGraphModifiedTime } from 'app/parser/util/VersionParser';
+
+abstract class AbstractLabradorParser extends AbstractIteratingParser {
+
+	// Implement AbstractParser
+
+	protected parseVersion() : Promise <string> {
+		return Promise.resolve(getOpenGraphModifiedTime(this.select));
+	}
+
+	protected parseByline() : Promise <ArticleAuthor[]> {
+		return Promise.resolve(getOpenGraphAuthors(this.select));
+	}
+
+	// Implement AbstractIteratingParser
+
+	protected getArticleContentScope() : string {
+		return 'main[role="main"]';
```

Comment:
Is this attribute role needed here? 
Because you rewrite the method in Sol parser just because Sol doesn't have it in html
Think we can live without it

Reply:
this role is present on the page, I guess we should define elements as accuracy as possible. It is not good that Sol does not have any other properties helping is ti define element. only css class, but strange - Main-s6qd7ta-0 cEkayc